ddtrace/span.py

```diff
@@ -226,7 +226,14 @@ def set_traceback(self):
             >>> span.set_exc_info(*exc)
         """
         (exc_type, exc_val, exc_tb) = sys.exc_info()
-        self.set_exc_info(exc_type, exc_val, exc_tb)
+
+        if (exc_type and exc_val and exc_tb):
+            self.set_exc_info(exc_type, exc_val, exc_tb)
+        else:
+            buff = StringIO()
+            traceback.print_stack(file=buff, limit=limit)
+            tb = buff.getvalue()
```

Comment:
instead of initializing a `StringIO()` file pointer (in-memory), I think you can just:
```python
tb = ''.join(traceback.format_stack(limit=limit))
self.set_tag(errors.ERROR_STACK, tb)
```